react-components/src/components/Architecture/pointsOfInterest/PoiList.tsx

```diff
@@ -1,44 +1,55 @@
 /*!
  * Copyright 2024 Cognite AS
  */
-import { type ReactNode, useMemo, useState } from 'react';
+import { type ReactNode, useCallback, useMemo, useState } from 'react';
 import { type PointOfInterest } from '../../../architecture/concrete/pointsOfInterest/types';
 import { useOnUpdateDomainObject } from '../useOnUpdate';
 import { DataGrid, type DatagridColumn } from '@cognite/cogs-lab';
 import { usePoiDomainObject } from './usePoiDomainObject';
 import { EMPTY_ARRAY } from '../../../utilities/constants';
 import { useTranslation } from '../../i18n/I18n';
+import { Button, Flex } from '@cognite/cogs.js';
+import { take } from 'lodash';
 
 type RowType = {
   id: string;
   poi: PointOfInterest<unknown>;
 };
 
+export type PoiFilter = { contains: string };
+
 export type PoiListProps = {
   onRowClick?: (poi: PointOfInterest<unknown>) => void;
+  filter?: PoiFilter;
+  pageLimit?: number;
 };
 
-export const PoiList = ({ onRowClick }: PoiListProps): ReactNode => {
+export const PoiList = ({ onRowClick, filter, pageLimit = Infinity }: PoiListProps): ReactNode => {
   const { t } = useTranslation();
 
   const [pois, setPois] = useState<Array<PointOfInterest<unknown>>>([]);
   const [selectedPoi, setSelectedPoi] = useState<PointOfInterest<unknown> | undefined>(undefined);
 
   const poiObject = usePoiDomainObject();
 
+  const [currentLimit, setCurrentLimit] = useState<number>(pageLimit ?? Infinity);
```

Comment:
Do we need `Infinity` here as default value is `Infinity`?

Reply:
Aah, woops :sweat_smile: